shared/teams/team/selection-popup.tsx

```diff
@@ -96,36 +92,70 @@ const SelectionPopup = (props: Props) => {
       </Kb.Text>
 
       {!Styles.isMobile && <Kb.BoxGrow />}
-      {props.selectedTab == 'channels' && (
-        <Kb.Button label="Delete" type="Danger" onClick={onDelete} fullWidth={Styles.isMobile} />
-      )}
 
-      {props.selectedTab == 'members' && (
-        <Kb.Box2
-          fullWidth={Styles.isMobile}
-          direction={Styles.isMobile ? 'vertical' : 'horizontal'}
-          gap="tiny"
-        >
-          <Kb.Button
-            label="Add to channels"
-            mode="Secondary"
-            onClick={onAddToChannel}
-            fullWidth={Styles.isMobile}
-          />
-          <Kb.Button label="Edit role" mode="Secondary" onClick={onEditRoles} fullWidth={Styles.isMobile} />
-          <Kb.Button
-            label="Remove from team"
-            type="Danger"
-            onClick={onRemoveFromTeam}
-            fullWidth={Styles.isMobile}
-          />
-        </Kb.Box2>
-      )}
+      <Actions teamID={props.teamID} />
     </Kb.Box2>
   ) : null
 }
 
-export default SelectionPopup
+type ActionsProps = {teamID: Types.TeamID}
+const ActionsWrapper = ({children}) => (
+  <Kb.Box2 fullWidth={Styles.isMobile} direction={Styles.isMobile ? 'vertical' : 'horizontal'} gap="tiny">
+    {children}
+  </Kb.Box2>
+)
+const MembersActions = ({teamID}: ActionsProps) => {
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const members = Container.useSelector(s => s.teams.selectedMembers.get(teamID))
+  if (!members) {
+    // we shouldn't be rendered
+    return null
+  }
+
+  // Members tab functions
+  const onAddToChannel = () =>
+    dispatch(
+      nav.safeNavigateAppendPayload({
+        path: [{props: {teamID, usernames: [...members]}, selected: 'teamAddToChannels'}],
+      })
+    )
+  const onEditRoles = () => {}
+  const onRemoveFromTeam = () => {}
+
+  return (
+    <ActionsWrapper>
+      <Kb.Button
```

Comment:
can we only show this button when there are channels?

Reply:
Yes. we should discuss a more reliable way of knowing whether a team is big in the gui.